### ISSUE TITLE
Is the masking correct for the NLL Loss?

### ISSUE THREAD

**Issue by fawazsammani** [ywk991112/pytorch-chatbot#12]
Hi. During the implementation of your code, I realized something. 
You may refer to the attached image for my question. 
There is a new added line I put which is: 
mask = mask.unsqueeze(1)
If you don't insert this line, the loss will be calculated just like without masking. It will simply take all the elements in the crossEntropyLoss (the 0s and 1s)
Because when masking, the two tensors have to be of same dimension according to the PyTorch documentation. So when you are masking a 2D tensor (the crossEntropyLoss) with a 1D tensor (the mask), then the mean of the unmasked and masked will have the same value. 

May you please clarify on this? 
![capture](https://user-images.githubusercontent.com/30661597/46492899-c0ae5200-c7c3-11e8-803d-58ea695d066f.PNG)

**Comment by ywk991112** [comment]
I change the loss function in this commit. [commit add29f](https://github.com/ywk991112/pytorch-chatbot/commit/add29fd92224d21e54e7feb895bc7dbdda75417a)
